tpu-client-next/src/workers_cache.rs

```diff
@@ -255,3 +311,149 @@ pub fn maybe_shutdown_worker(worker: Option<ShutdownWorker>) {
         }
     });
 }
+
```

Comment:
Lets add tests in separate PR this is already huge

Reply:
this pr is on top of https://github.com/anza-xyz/agave/pull/5888, so this one contains almost only tests.